pkg/controller/test/context_builder.go

```diff
@@ -148,6 +152,7 @@ func (b *Builder) Init() {
 	b.FakeKubeClient().PrependReactor("create", "*", b.generateNameReactor)
 	b.FakeCMClient().PrependReactor("create", "*", b.generateNameReactor)
 	b.FakeGWClient().PrependReactor("create", "*", b.generateNameReactor)
+	b.FakeMetadataClient().PrependReactor("create", "*", b.generateNameReactor)
```

Comment:
This line was missing.

Reply:
I don't understand what bug is being fixed. 

`FakeMetadataClient()` and `FakeDiscoveryClient()` are added in this PR so it looks more like new features being added.

And why isn't there a similar new line, like `FakeDiscoveryClient().PrependReactor(...` ?

Reply:
`FakeDiscoveryClient` does not have a `PrependReactor` function, that is why I did not add it here.
The FakeMetadataClient() and FakeDiscoveryClient() functions were indeed added in this PR, but the MetadataClient and DiscoveryClient fields already existed before.
It's a but because the reactor was not added, but I don't think it altered the behavior for any of the existing tests, but might prevent issues in the future.